lib/new_relic/cli/commands/deployments.rb

```diff
@@ -109,22 +109,76 @@ def run
     end
   end
 
+  def api_v1?
+    @api_key.nil? || @api_key.empty?
+  end
+
   private
 
+  def create_request(uri, headers, content_type)
+    Net::HTTP::Post.new(uri, headers).tap do |req|
+      req.content_type = content_type
+    end
+  end
+
+  def application_id
+    return @application_id if @application_id
+
+    # Need to connect to collector to acquire application id from the connect response
+    # but set monitor_mode false because we don't want to actually report anything
+    begin
+      NewRelic::Agent.manual_start(monitor_mode: false)
+      NewRelic::Agent.agent.connect_to_server
+      @application_id = NewRelic::Agent.config[:primary_application_id]
+    ensure
+      NewRelic::Agent.shutdown
+    end
+  end
+
+  def set_params_v1(request)
+    params = {}
+    {
+      :application_id => @appname,
+      :host => NewRelic::Agent::Hostname.get,
+      :description => @description,
+      :user => @user,
+      :revision => @revision,
+      :changelog => @changelog
+    }.each do |k, v|
+      params["deployment[#{k}]"] = v unless v.nil? || v == ''
+    end
```

Comment:
You could use `#each_with_object` here. I'm not really sure what it does for object allocation and performance compared to manually starting with an empty hash as a value for a var, so definitely consider this an optional-to-implement suggestion. It's just how I would have written it.

```diff
- params = {}
- {
+ params = {
    :application_id => @appname,
    :host => NewRelic::Agent::Hostname.get,
    :description => @description,
    :user => @user,
    :revision => @revision,
    :changelog => @changelog
- }.each do |k, v|
+ }.each_with_object({}) do |(k, v), h|
    params["deployment[#{k}]"] = v unless v.nil? || v == ''
  end
```

Reply:
oh yeah that makes sense. I didn't change this chunk much other than just moving it around, but I can do that!

Reply:
addressed 5b7e76cc0f08701e2b5da74305327b312c74c59b

Reply:
Whoops! Sorry for missing the `params` -> `h` change that needed to be a part of the diff here. The way it looks in your code is great. Thanks for addressing.